addons/mail/models/mail_channel.py

```diff
@@ -280,6 +286,15 @@ def write(self, vals):
             if not self.env.user.has_group('base.group_system'):
                 raise UserError(_("You do not have the rights to modify fields related to moderation on one of the channels you are modifying."))
 
+        # If users tries to remove themselves from 'Members' of a private channel, read access will be revoked
+        # immediately and they won't be able to save the changes due to ACL warning. In this case, simply ask
+        # them to leave the channel from the header button.
+        if len(self) == 1 and self.public == 'private' and vals.get('channel_last_seen_partner_ids'):
```

Comment:
It would be nice to support the batch mode of the method write (to be future proof)

Also, can you add a test about the fix ?